### DISCUSSION TITLE
how about to run mirth connectoin on cockroach db?

### DISCUSSION THREAD

**Discussion by linxd** [nextgenhealthcare/connect#5641] [Feature/Enhancement Ideas]
I have do some test to run MC on cockroach db , and found that MC can run on cockroach  allmost  with no problem. someone has done this or have some suggestions to exchange?

just change the database.url  setting, other same with postgresql driver. 

Thanks.

**Comment by jonbartels** [comment]
Interesting finding. 

Mirth Connect *should* be compatible with any SQL-based relational database which has a JDBC driver.

The place to look is at the mybatis XML scripts which show the queries for `default` and for other specific database engines. A full integration with CockroachDB would require ONE OF:
* verifying that CoachroachDB is compatible with an existing DB mapping for another DB engine or the default 
* writing custom DB mappings for CockroachDB and re-compiling Mirth Connect to add this support for any Cockroach specific items

Skimming Cockroach documentation:
* Basics - https://github.com/cockroachdb/cockroach
* SQL - https://www.cockroachlabs.com/docs/v22.2/sql-statements ; This looks like standard SQL-99 SQL

**Reply by jonbartels** [reply]
Reading https://www.cockroachlabs.com/docs/v22.2/postgresql-compatibility.html which says:

> CockroachDB supports the [PostgreSQL wire protocol](https://www.postgresql.org/docs/current/protocol.html) and the majority of PostgreSQL syntax. This means that existing applications built on PostgreSQL can often be migrated to CockroachDB without changing application code.
> 
> CockroachDB is compatible with version 3.0 of the PostgreSQL wire protocol (pgwire) and works with the majority of PostgreSQL database tools such as [DBeaver](https://www.cockroachlabs.com/docs/v22.2/dbeaver), [Intellij](https://www.cockroachlabs.com/docs/v22.2/intellij-idea), and so on. Consult this link for a full list of supported [third-party database tools](https://www.cockroachlabs.com/docs/v22.2/third-party-database-tools). CockroachDB also works with most [PostgreSQL drivers and ORMs](https://www.cockroachlabs.com/docs/v22.2/example-apps).
> 
> However, CockroachDB does not support some of the PostgreSQL features or behaves differently from PostgreSQL because not all features can be easily implemented in a distributed system. This page documents the known list of differences between PostgreSQL and CockroachDB for identical input. That is, a SQL statement of the type listed here will behave differently than in PostgreSQL. Porting an existing application to CockroachDB will require changing these expressions.

A close review of the incompatible features against the Mirth Connect postgres mapping should show if MC will work with Cockroach without modification.
